frontend/app/utils/adobe-analytics.client.ts

```diff
@@ -24,16 +24,22 @@ export const pageview = (locationUrl: string) => {
   }
 
   appPageLoadPreviousLocationPathname = locationUrlObj.pathname;
+  let url = `${locationUrlObj.host}${locationUrlObj.pathname}`;
 
   // Adobe Analytics needs us to clean up URLs before sending event data. This ensures their reports focus
   // on the core content, not things like tracking codes, because they don't want those to mess up their
   // website visitor categories.
-  const transformedUrl = removeUUIDSegmentsFromURL(locationUrlObj);
+  if (
+    ['/en/apply/', '/fr/apply/', '/en/appliquer/', '/fr/appliquer/'].some((str) => {
+      return locationUrlObj.pathname.toLocaleLowerCase().startsWith(str);
+    })
+  ) {
+    const transformedUrl = removeUUIDSegmentsFromURL(locationUrlObj);
+    url = `${transformedUrl.host}${transformedUrl.pathname}`;
+  }
```

Comment:
I don't like that the URL removal code is so specific.. like mentioned in slack, if the user changes the UUID in the address to something that isn't valid (ex: removes a hyphen), then the `removeUUIDSegmentsFromURL()` call will not work.

Also, since this code is specific to the apply flow, it should be colocated with it. This code isn't globally usable, so there's no reason to put it in a util module.

My suggestion would be to move the code into the /appy/ route, and have it remove the flow ID by doing something like `replace(/(en|fr)\/(apply|appliquer)\/.+(\/.*)/$1$2$3)`

Reply:
I completely revamped that code. Please review again.